pkg/agent/openflow/pod_connectivity.go

```diff
@@ -157,16 +163,105 @@ func (f *featurePodConnectivity) initFlows() []binding.Flow {
 		// Pod IP will take care of routing the traffic to destination Pod.
 		flows = append(flows, f.l3FwdFlowToLocalPodCIDR()...)
 	}
+	if f.enableTrafficControl {
+		flows = append(flows, f.redirectedPktsFwdFlows()...)
+	}
 	return flows
 }
 
 func (f *featurePodConnectivity) replayFlows() []binding.Flow {
 	var flows []binding.Flow
 
 	// Get cached flows.
-	for _, cachedFlows := range []*flowCategoryCache{f.nodeCachedFlows, f.podCachedFlows} {
+	for _, cachedFlows := range []*flowCategoryCache{f.nodeCachedFlows, f.podCachedFlows, f.tcCachedFlows} {
 		flows = append(flows, getCachedFlows(cachedFlows)...)
 	}
 
 	return flows
 }
+
+// trafficControlMarkFlows generates the flows to mark the packets that need to be redirected or mirrored with the value of
+// the destination output port.
+func (f *featurePodConnectivity) trafficControlMarkFlows(srcOfPorts []uint32, dstOfPort uint32, direction v1alpha2.Direction) []binding.Flow {
+	cookieID := f.cookieAllocator.Request(f.category).Raw()
+	var flows []binding.Flow
+	for _, port := range srcOfPorts {
+		if direction == v1alpha2.DirectionIn || direction == v1alpha2.DirectionBoth {
+			flows = append(flows, IngressTrafficControlMarkTable.ofTable.BuildFlow(priorityNormal).
+				Cookie(cookieID).
+				MatchRegFieldWithValue(TargetOFPortField, port).
+				Action().LoadToRegField(TrafficControlOFPortField, dstOfPort).
+				Action().NextTable().
+				Done())
+		}
+		if direction == v1alpha2.DirectionOut || direction == v1alpha2.DirectionBoth {
+			flows = append(flows, EgressTrafficControlMarkTable.ofTable.BuildFlow(priorityNormal).
+				Cookie(cookieID).
+				MatchInPort(port).
+				Action().LoadToRegField(TrafficControlOFPortField, dstOfPort).
+				Action().NextTable().
+				Done())
+		}
+	}
+	return flows
+}
+
+// trafficControlInitFlows generates the initial flows for the traffic control mark value.
```

Comment:
What does "for the traffic control mark value" mean? Consider rephrasing.

Reply:
The updated comment: `trafficControlInitFlows generates the initial flows of traffic control to forward the marked packets and the redirected packets.`

Reply:
Ok. Probably do not say "initial flows" but "common flows", and rename the func to trafficControlCommonFlows() - it is not about "initialization".